tests/basic/customer_issues.go

```diff
@@ -0,0 +1,352 @@
+			}
+		}
+
+		// Create 2 zones from the Storage and storageless Node
+		zone1 := []node.Node{}
+		zone2 := []node.Node{}
+
+		zone1StorageEle, zone2StorageEle := getRandomNumbersFromArrLength(len(allStorageExceptKVDB), len(allStorageExceptKVDB)/2)
+		for _, each := range zone1StorageEle {
+			zone1 = append(zone1, allStorageExceptKVDB[each])
+		}
+		for _, each := range zone2StorageEle {
+			zone2 = append(zone2, allStorageExceptKVDB[each])
+		}
+
+		zone1StorageLessEle, zone2StorageLessEle := getRandomNumbersFromArrLength(len(nodesSplit2), len(nodesSplit2)/2)
+		for _, each := range zone1StorageLessEle {
+			zone1 = append(zone1, nodesSplit2[each])
+		}
+		for _, each := range zone2StorageLessEle {
+			zone2 = append(zone2, nodesSplit2[each])
+		}
+
+		flushiptables := func() {
+			if !iptablesflushed {
+				// Flush all iptables from all the nodes available forcefully
+				for _, eachNode := range allNodes {
+					log.InfoD("Flushing iptables rules on node [%v]", eachNode.Name)
+					log.FailOnError(flushIptableRules(eachNode), "Iptables flush all failed on node [%v]", eachNode.Name)
+				}
+			}
+		}
+
+		// force flush iptables on all the nodes at the end
+		defer flushiptables()
+
+		volumeState, err := GetVolumesInDegradedState(contexts)
+		if err != nil {
+			fmt.Println("Error not Nil")
+		}
+		fmt.Println(volumeState)
+
+		revertZone1 := func() {
+			log.FailOnError(blockIptableRules(zone1, zone2, true), "Failed to unblock IPTable rules on target Nodes")
+		}
+
+		revertZone2 := func() {
+			log.FailOnError(blockIptableRules(zone2, zone1, true), "Failed to unblock IPTable rules on target Nodes")
+		}
+
+		// From Zone 1 block all the traffic to systems under zone2
+		// From Zone 2 block all the traffic to systems under zone1
+		log.InfoD("blocking iptables from all nodes present in zone1 from accessing zone2")
+		err = blockIptableRules(zone1, zone2, false)
+		log.FailOnError(err, "Failed to revert IPtable Rules on Zone1")
+
+		time.Sleep(20 * time.Minute)
+
+		log.InfoD("blocking iptables from all nodes present in zone2 from accessing zone1 ")
+		err = blockIptableRules(zone2, zone1, false)
+		log.FailOnError(err, "Failed to set IPtable Rules on zone2")
+
+		// Reverting back Zone1 iptables set
+		revertZone1()
+
+		// Reverting back zone2 iptables set
+		revertZone2()
+
+		// Reset iptables rules on vms under zone2
+		err = blockIptableRules(zone2, zone1, false)
+		log.FailOnError(err, "Failed to set IPtable Rules on zone2")
+
+		log.InfoD("Killing KVDB PID from KVDB Master Node")
+		log.FailOnError(KillKvdbNode(getKvdbLeaderNode), "failed to Kill Kvdb Node")
+		// Block IPtable rules on the kvdb node to all the nodes in zone 1
+		kvdb := []node.Node{getKvdbLeaderNode}
+
+		time.Sleep(10 * time.Minute)
+		log.FailOnError(blockIptableRules(kvdb, zone1, false), "Set IPTable rules on kvdb node failed")
+
+		// Wait for some time before checking for file system goes back online
+		time.Sleep(30 * time.Minute)
```

Comment:
add log statments for these long waits. like how much time you are waiting and why are you waiting

Reply:
done